app/code/Magento/Inventory/Model/GetProductQuantityInStock.php

```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Inventory\Model;
+
+use Magento\Inventory\Model\ResourceModel\Reservation\ReservationQuantity;
+use Magento\Inventory\Model\ResourceModel\Stock\StockItemQuantity;
+use Magento\InventoryApi\Api\GetProductQuantityInStockInterface;
+
+/**
+ * Return Quantity of products available to be sold by Product SKU and Stock Id
+ *
+ * @see \Magento\InventoryApi\Api\GetProductQuantityInStockInterface
+ * @api
+ */
+class GetProductQuantityInStock implements GetProductQuantityInStockInterface
+{
+    /**
+     * GetProductQuantityInStock constructor.
+     *
+     * @param StockItemQuantity $stockItemQty
+     * @param ReservationQuantity $reservationQty
+     */
+    public function __construct(
+        StockItemQuantity $stockItemQty,
+        ReservationQuantity $reservationQty
+    ) {
+        $this->stockItemQty = $stockItemQty;
+        $this->reservationQty = $reservationQty;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function execute(string $sku, int $stockId): float
+    {
+        $productQtyInStock = $this->stockItemQty->execute($sku, $stockId) -
+            $this->reservationQty->execute($sku, $stockId);
```

Comment:
if we are putting both positive and negative values to the Reservation.
Looks like we need to **Add** `$this->reservationQty->execute($sku, $stockId);`
But not **Substract**

For example, 
Stock is 100 products.
We have Reservation for -10 products 

`100  - (-10)` will lead to increase of stock but not deducation

Reply:
I've just re-read your spec [here](https://github.com/magento-engcom/msi/wiki/Reservations#removal-of-processed-reservations) and you were perfectly clear on the meaning of adding (+) and subtracting (-) but for some reason, I inverted it in my mind